api/v1beta1/gcpmachine_types.go

```diff
@@ -66,6 +66,25 @@ const (
 	IPForwardingDisabled IPForwarding = "Disabled"
 )
 
+// AcceleratorConfig is the GPU accelerator configuration for the GCP machine.
+type AcceleratorConfig struct {
+	// Type is the type of the GPU accelerator to be used for the GCP machine.
+	// +required
+	Type string `json:"acceleratorType"`
+
+	// Count is the number of accelerators to be used for the GCP machine.
+	// Defaults to 1.
+	// +optional
+	Count int64 `json:"acceleratorCount,omitempty"`
```

Comment:
We can probably use `// +kubebuilder:default:=1` here.

Reply:
For the current case the validation of accelerators during the instance creation does not allow `int` to be used, further if we change the static validation function to a `conditional check` during reconciliation we can use `int`.

Reply:
As a principle, our API doesn't have to match the GGCP exactly, so we have use different data types if needed (as long as we can caste without loosing data).